packages/taquito-sapling/src/taquito-sapling.ts

```diff
@@ -3,6 +3,323 @@
+    const { formatedParams, totalAmount } = this.formatTransactionParams(
+      [unshieldedTxParams],
+      this.validateDestinationImplicitAddress
+    );
+
+    const boundData = await this.createBoundData(formatedParams[0].to);
+
+    const root = await this.getRoot();
+    const chosenInputs = await this.selectInputsToSpend(new BigNumber(formatedParams[0].amount));
+
+    const { inputs, outputs, signature, balance } = await this.#saplingTxBuilder.createSaplingTx(
+      [],
+      totalAmount,
+      boundData,
+      chosenInputs
+    );
+
+    const forgedSaplingTx = this.#saplingForger.forgeSaplingTransaction({
+      inputs,
+      outputs,
+      balance,
+      root,
+      boundData,
+      signature,
+    });
+
+    return forgedSaplingTx.toString('hex');
+  }
+
+  /**
+   * @description Prepare a sapling transaction (zet to zet)
+   * @param saplingTxParams `to` is the payment address that will receive the shielded tokens (zet).
+   * `amount` is the amount of unshielded tokens in tez by default.
+   * `mutez` needs to be set to true if the amount of unshielded tokens is in mutez.
+   * `memo` is an empty string by default.
+   * @returns a string representing the sapling transaction.
+   */
+  async prepareSaplingTransaction(saplingTxParams: ParametersSaplingTransaction[]) {
+    const { formatedParams, totalAmount } = this.formatTransactionParams(
+      saplingTxParams,
+      this.validateDestinationSaplingAddress
+    );
+
+    const root = await this.getRoot();
+    const chosenInputs = await this.selectInputsToSpend(totalAmount);
+    const { inputs, outputs, signature, balance } = await this.#saplingTxBuilder.createSaplingTx(
+      formatedParams,
+      totalAmount,
+      defaultBoundData,
+      chosenInputs
+    );
+
+    const forgedSaplingTx = this.#saplingForger.forgeSaplingTransaction({
+      inputs,
+      outputs,
+      balance,
+      root,
+      boundData: defaultBoundData,
+      signature,
+    });
+
+    return forgedSaplingTx.toString('hex');
+  }
+
+  private formatTransactionParams(
+    txParams: ParametersSaplingTransaction[],
+    validateDestination: (to: string) => void
+  ) {
+    const formatedParams: {
+      to: string;
+      amount: string;
+      memo: string;
+    }[] = [];
+
+    let totalAmount = new BigNumber(0);
+
+    txParams.forEach((param) => {
+      validateDestination(param.to);
+      const amountMutez = param.mutez
+        ? param.amount.toString()
+        : format('tz', 'mutez', param.amount).toString();
+      totalAmount = totalAmount.plus(new BigNumber(amountMutez));
+      const memo = param.memo ?? defaultMemo;
+      if (memo.length > this.#memoSize) {
+        throw new InvalidMemo(memo, 'The memo is too long.');
+      }
+
+      formatedParams.push({ to: param.to, amount: amountMutez, memo });
+    });
+
+    return { formatedParams, totalAmount };
+  }
+
+  private async getRoot() {
```

Comment:
is this guaranteed to return a root? or is it possible to be undefined? should we handle the potential undefined is so?

Reply:
I don't think the root can be undefined.. even when there is no transactions yet in the pool, there is a root